client/asset/dcr/dcr.go

```diff
@@ -3498,52 +3498,62 @@ func (dcr *ExchangeWallet) monitorBlocks(ctx context.Context) {
 	// notification.
 	var queuedBlock *polledBlock
 
-	for {
-		select {
-		case <-ticker.C:
-			ctx, cancel := context.WithTimeout(dcr.ctx, 2*time.Second)
-			defer cancel()
-			newTip, err := dcr.getBestBlock(ctx)
-			if err != nil {
-				dcr.handleTipChange(nil, 0, fmt.Errorf("failed to get best block: %w", err))
-				return
```

Comment:
This return was the bug that resulted in "client has been shutdown" and unable to reconnect to the wallet without dexc restart.  The `defer cancel()` was also not working as intended, but also filling up the defer stack leading to memory exhaustion.